torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -472,19 +466,16 @@ void ProcessGroupAgent::pollTimedOutRPCs() {
     // for that long.
     // if there are no futures or the first future's RPC timeout is set to 0
     // (meaning no timeout), then sleep for a set "infinity" time.
-    if (futureTimeouts_.empty() ||
-        futureTimeouts_.begin()->first == INFINITE_TIMEOUT) {
-      sleepTime = INFINITE_TIMEOUT;
+    if (futureTimeouts_.empty()) {
+      minEndTime = kInfiniteTimeoutTimePoint;
     } else {
-      const auto minFutureExpirationTime = futureTimeouts_.begin()->first;
-      const auto remainingTime = getRPCRemainingTime(minFutureExpirationTime);
-      sleepTime = std::max(remainingTime, std::chrono::milliseconds(0));
+      minEndTime = futureTimeouts_.begin()->first;
     }
 
-    if (sleepTime == INFINITE_TIMEOUT) {
+    if (minEndTime == kInfiniteTimeoutTimePoint) {
       futureTimeoutCV_.wait(lock);
     } else {
-      futureTimeoutCV_.wait_for(lock, sleepTime);
+      futureTimeoutCV_.wait_until(lock, minEndTime);
```

Comment:
I think you can get rid of the `sleepTime` variable now?

Reply:
Right. Let me update.